src/lib/support/BUILD.gn

```diff
@@ -127,3 +131,7 @@ copy("support_headers") {
 
   outputs = [ "${root_gen_dir}/include/support/{{source_file_part}}" ]
 }
+
+config("c_includes") {
+  include_dirs = [ "c" ]
```

Comment:
What's the separate dir for?

Reply:
I can't get GN to build it otherwise — include_dir'ing `support` or any higher directory triggers an error about `BLEConfig.h` including `CHIPConfig.h`.

Reply:
This is because of a preexisting cycle, to fix:

```
diff --git a/src/lib/core/BUILD.gn b/src/lib/core/BUILD.gn
index ee2156e1..f38d1ac4 100644
--- a/src/lib/core/BUILD.gn
+++ b/src/lib/core/BUILD.gn
@@ -84,6 +84,10 @@ source_set("chip_config_header") {
     "${chip_root}/src/ble:ble_config_header",
     "${chip_root}/src/system:system_config_header",
   ]
+
+  allow_circular_includes_from = [
+    "${chip_root}/src/ble:ble_config_header",
+  ]
 }
 
 static_library("core") {

```